MapboxNavigationTests/CarPlayManagerTests.swift

```diff
@@ -170,8 +169,35 @@ class CarPlayManagerTests: XCTestCase {
 
         XCTAssertTrue(exampleDelegate.navigationEnded, "The CarPlayManagerDelegate should have been told that navigation ended.")
     }
+    func testDirectionsOverride() {
+        let expectation = XCTestExpectation(description: "Ensuring Spy is called")
+        let spy = DirectionsInvocationSpy(accessToken: "DeadBeefCafe", host: nil)
+        spy.payload = expectation.fulfill
+        
+        let subject = CarPlayManager(directions: spy)
+      
+        let waypoint1 = Waypoint(coordinate: CLLocationCoordinate2D(latitude: 37.795042, longitude: -122.413165))
+        let waypoint2 = Waypoint(coordinate: CLLocationCoordinate2D(latitude: 37.7727, longitude: -122.433378))
+        let options = RouteOptions(waypoints: [waypoint1, waypoint2])
+        subject.fetchNewRoute(options: options, completionHandler: {})
+        wait(for: [expectation], timeout: 1.0)
+        
+        XCTAssert(subject.directions == spy, "Directions client is not overridden properly.")
+    }
 }
 
+private class DirectionsInvocationSpy: Directions {
```

Comment:
This spy class is very specific to a single test. Consider moving this class definition inside `testDirectionsOverride()` so the relationship is clearer.